core/go/internal/groupmgr/rpc_eventstreams.go

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright © 2024 Kaleido, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package groupmgr
+
+import (
+	"context"
+	"sync"
+
+	"github.com/kaleido-io/paladin/core/internal/components"
+	"github.com/kaleido-io/paladin/core/internal/msgs"
+	"github.com/kaleido-io/paladin/toolkit/pkg/i18n"
+	"github.com/kaleido-io/paladin/toolkit/pkg/log"
+	"github.com/kaleido-io/paladin/toolkit/pkg/pldapi"
+	"github.com/kaleido-io/paladin/toolkit/pkg/rpcclient"
+	"github.com/kaleido-io/paladin/toolkit/pkg/rpcserver"
+	"github.com/kaleido-io/paladin/toolkit/pkg/tktypes"
+)
+
+type rpcEventStreams struct {
+	gm          *groupManager
+	subLock     sync.Mutex
+	receiptSubs map[string]*receiptListenerSubscription
+}
+
+func newRPCEventStreams(tm *groupManager) *rpcEventStreams {
+	es := &rpcEventStreams{
+		gm:          tm,
+		receiptSubs: make(map[string]*receiptListenerSubscription),
+	}
+	return es
+}
+
+func (es *rpcEventStreams) StartMethod() string {
+	return "pgroup_subscribe"
+}
+
+func (es *rpcEventStreams) LifecycleMethods() []string {
+	return []string{"pgroup_unsubscribe", "pgroup_ack", "pgroup_nack"}
+}
+
+type rpcAckNack struct {
+	ack bool
+}
+
+type receiptListenerSubscription struct {
+	es        *rpcEventStreams
+	pgmrc     components.PrivacyGroupMessageReceiverCloser
+	ctrl      rpcserver.RPCAsyncControl
+	acksNacks chan *rpcAckNack
+	closed    chan struct{}
+}
+
+func (es *rpcEventStreams) HandleStart(ctx context.Context, req *rpcclient.RPCRequest, ctrl rpcserver.RPCAsyncControl) (rpcserver.RPCAsyncInstance, *rpcclient.RPCResponse) {
+	es.subLock.Lock()
+	defer es.subLock.Unlock()
+
+	var eventType tktypes.Enum[pldapi.PGroupEventType]
+	if len(req.Params) >= 1 {
+		eventType = tktypes.Enum[pldapi.PGroupEventType](req.Params[0].StringValue())
```

Comment:
Could maybe be clearer about these params... I'm seeing that:
- `req.Params[0]` is the event type (currently only one type "messages" supported)
- for type "messages", there should be exactly one additional parameter, and that is the listener name

But it took a few minutes of searching around to infer that.

Reply:
Understood. I've put together some docs, but event docs could be more comprehensive:

https://lf-decentralized-trust-labs.github.io/paladin/head/reference/types/transactionreceiptlisteneroptions/#create-receipt-listener (which is moving to `CreateReceiptListener` in this PR)